code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -154,6 +159,35 @@
 	else
 		to_chat(src, span_warning("You feel fine, no need to eat anything!"))
 
+/**
+ * Allows rat king to pry open an airlock if it isn't locked.
+ *
+ * A proc used for letting the rat king pry open airlocks instead of just attacking them.
+ * This allows the rat king to traverse the station when there is a lack of vents or
+ * accessible doors, something which is common in certain rat king spawn points.
+ */
+/mob/living/simple_animal/hostile/regalrat/proc/pry_door(target)
+	if(opening_airlock)
+		return FALSE
+	opening_airlock = TRUE
+	var/obj/machinery/door/airlock/prying_door = target
+	if(prying_door.density && !(prying_door.locked || prying_door.welded || prying_door.seal))
+		visible_message(span_warning("[src] begins prying open the airlock..."),\
+			span_notice("You begin digging your claws into the airlock..."),\
+			span_warning("You hear groaning metal..."))
+		var/time_to_open = 0.5 SECONDS
+		if(prying_door.hasPower())
+			time_to_open = 5 SECONDS
+			playsound(src, 'sound/machines/airlock_alien_prying.ogg', 100, TRUE)
+		if(do_after(src, time_to_open, prying_door))
+			opening_airlock = FALSE
+			if(prying_door.density && !prying_door.open(2))
```

Comment:
Named parameter for `2`

Reply:
Though tbh it looks like a magic number? Not sure

Reply:
This is a magic number as far as I'm aware.